File: 56. Merge Intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        merged = [intervals[0]]
        if len(intervals) <= 1:
            return intervals
        else:
            for i in range(1,len(intervals)):
                if merged[-1][1] >= intervals[i][0]:
                    if merged[-1][1] < intervals[i][1]:
                        merged[-1][1] = intervals[i][1]
                else:
                    merged.append(intervals[i])
        return merged

File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> res;
        sort(intervals.begin(), intervals.end());
        for (auto i: intervals) {
            if (res.empty() || res[res.size()-1][1] < i[0])
                res.push_back(i);
            else {
                res[res.size()-1][1] = (res[res.size() - 1][1] > i[1]? res[res.size() - 1][1]: i[1]);
            }
        }
        return res;
    }
};